### ISSUE TITLE
Broken getProvider method

### ISSUE THREAD

**Issue by a-komarev** [geocoder-php/GeocoderLaravel#99]
It's trying to access to private method:

https://github.com/geocoder-php/GeocoderLaravel/blob/b424b36bd6350408fa9609dd0fbbc2d7118c5654/src/ProviderAndDumperAggregator.php#L181-L184

https://github.com/geocoder-php/Geocoder/blob/69167527c25779e604791f5eef4f3f9a96b6e9dc/src/Common/ProviderAggregator.php#L166-L192

**Comment by mikebronner** [comment]
Why is this useless? Please provide more detail as to what the problem is, what you are trying to do, and what you think the solution would be.

**Comment by a-komarev** [comment]
Because you cannot call private method `getProvider` outside of the class, there will be an exception.

**Comment by mikebronner** [comment]
There is no problem that I have seen. I have unit tests for all these methods. Please provide a complete stack trace, package version, PHP version, Laravel version, and providers and aggregators you are using.

**Comment by a-komarev** [comment]
```
PHP 7.1
"toin0u/geocoder-laravel": "^4.0"
```

Code sample:
```php
$provider = app('geocoder')->getProvider();
dd($provider);
```

> Error: Call to protected method Geocoder\Laravel\ProviderAndDumperAggregator::getProvider()

Any version of PHP will throw an exception. because you cannot call private method outside of the class!

**Comment by mikebronner** [comment]
OK, thanks. I see it now. Use `->getProviders()`. Investigating if it needs to be removed or fixed.

**Comment by a-komarev** [comment]
Yep. You still can get provider from Address instance:

```php
$provider = app('geocoder)->geocode('Los Angeles, CA')->get()->first()->getProvidedBy();
```

**Comment by mikebronner** [comment]
I inspected how the parent package works, and it basically just returns the first provider from all available ones. Posting a fix today, but will continue looking into a way to get the provider used for the current result. :)

**Comment by mikebronner** [comment]
Was looking more into this and thinking about it. Providing it at a higher level actually provides very little, if any value, as the each result could be provided by a different provider, so you must inspect at the result level, then if no results, look at what providers were loaded:

```
$providers = app('geocoder)->geocode('Los Angeles, CA')->getProviders();
$result = app('geocoder')->get()->first();
$provider = $result->getProvider() ?? $providers;
```
or something similar.

**Comment by mikebronner** [comment]
Better to leave this to the user to implement, as this will probably rarely be used.

**Comment by a-komarev** [comment]
Is there a possibility to get collection of results from different providers? I thought that providers will be called one by one until first one wouldn't return the success result.

**Comment by mikebronner** [comment]
Sure, that would involve iterating over each result and aggregating the used providers, just the same. Will look into if that's easy to implement.

**Comment by a-komarev** [comment]
If there could be collection of results from different providers thats make sense then. There is no need to have it on the top level. I thought collection must contains results from one provider only.

**Comment by mikebronner** [comment]
oh, no, they are mixed, you get only one collection back

**Comment by mikebronner** [comment]
This is now "fixed" in 4.0.2 and marked as deprecated. Thanks for working through this with me.

**Comment by a-komarev** [comment]
You are welcome! Thank you for package support and the quick fix!
